src/components/notes/saving-indicator.jsx

```diff
@@ -2,28 +2,47 @@ import React, { useEffect, useCallback, useState } from 'react'
 import { useI18n } from 'cozy-ui/react/I18n'
 import { Text } from 'cozy-ui/react/Text'
 import usePeriodicRender from 'cozy-ui/react/hooks/usePeriodicRender'
+import useEventListener from 'cozy-ui/react/hooks/useEventListener'
+import useBreakpoints from 'cozy-ui/react/hooks/useBreakpoints'
 import styles from 'components/notes/saving-indicator.styl'
 import throttle from 'lodash/throttle'
 import { relativeAge } from 'lib/utils'
-
+import OfflineIndicator from 'components/notes/offline-indicator'
 // constants
 const sec = 1000
+const keepOfflineOpenFor = 5 * sec
+const outOfSyncAfter = 30 * sec
 
 /**
  * @typedef {object} SavingIndicatorProps
  * @property {CollabProviderDate} collabProvider
+ * @property {Ref} bannerRef - Where to display banners
  */
 
 /**
  * Displays a saving indicator in the bottom right
+ * and an offline indicator if needed
  *
  * @param {SavingIndicatorProps} props
  */
 function SavingIndicator(props) {
   const { t } = useI18n()
   const now = new Date()
+  const { isMobile } = useBreakpoints()
 
   const [, setLastChange] = useState(undefined)
+  const [offlineSince, setOfflineSince] = useState(undefined)
+  const [isNavigatorOffline, setIsNavigatorOffline] = useState(
+    window && window.navigator && window.navigator.onLine === false
+  )
+  const setNavigatorOffline = useCallback(() => setIsNavigatorOffline(true), [
+    setIsNavigatorOffline
+  ])
+  const setNavigatorOnline = useCallback(() => setIsNavigatorOffline(false), [
+    setIsNavigatorOffline
+  ])
+  useEventListener(window, 'online', setNavigatorOnline)
+  useEventListener(window, 'offline', setNavigatorOffline)
```

Comment:
Nit: I am wondering if some of the online/offline logic could be moved outside the component ? Maybe with a useNetworkStatus hook ? It could also take the offlineSince state part ?

Reply:
I don't think we'll really mutualize anything and it will add a custom extra layer upon something simple and already known